testng-core/src/test/groovy/test/groovy/SpockSample.groovy

```diff
@@ -1,9 +1,11 @@
 package test.groovy
 
+import org.junit.Test
 import spock.lang.Specification
 
 class SpockSample extends Specification {
 
+    @Test
```

Comment:
@juherr without the `@Test` annotation, JUnit refuses to find the method and crashes with `No runnable methods` error. I have tried this same sample and went back as old as https://mvnrepository.com/artifact/org.testng/testng/6.9.13.6 which was released in Sept 2016 and the behavior is still the same. If you know what is causing this problem, please help point to it. But without this annotation the test is not discoverable by JUnit.

Reply:
That's the point: a Spock test is not a JUnit test
There is a very specific integration https://github.com/cbeust/testng/blob/563bd6dfe2e74475b45110d1cea8101641ebf2b5/testng-core/src/main/java/org/testng/junit/JUnit4SpockMethod.java and the test was here to check the integration.

If the test is failing, it just means the integration is broken.

Reply:
@juherr - This class was created to deal with tests it looks like. I dont see any code which seems to suggest that it was created to discover tests. I say this because this class is getting instantiated ONLY when we are at the point of creating results.

`org.testng.junit.JUnit4TestRunner#createTestResult` > `org.testng.junit.JUnit4TestMethod#JUnit4TestMethod` > `org.testng.junit.JUnit4TestMethod#getMethod`

This is the call stack. I dont see any traces of code that attempts at discovering a Spock test in our codebase.

Reply:
Here's what I see as the exception 

<img width="1062" alt="image" src="https://user-images.githubusercontent.com/909773/205230657-9ab292e3-4293-4f11-9345-7e4267eea3da.png">

Reply:
Ignore all this. You were right. I found the problem. It was version upgrades :) We were trying to work with spock 2.x series which expects the JUnit engine like runner to run but TestNG is not going to be there :) so i had to pin us back to the 1.x series of version.

Reply:
Maybe we can deprecate the Spock support in the next major release.
In case users have both testng and spock tests, I think the current state of the art is using the junit5 platform with spock and testng flavors.

Reply:
In fact, same remark for the JUnit support

Reply:
So TestNG 8.0 with the deprecation or we deprecate in 7.7.0 and then get rid of the support for JUnit and spock in 8 ?